src/urbit/mar/chess/game.hoon

```diff
@@ -1,4 +1,4 @@
-/-  *chess
+/-  chess
```

Comment:
Just flagging this up for later: IIRC I wrote the `['moves' [%a ~]]` in this file because it was better to explicitly send `null` to the frontend than implicitly send nothing, but @ashelkovnykov wondered if it could be filled in to use this `%chess-game` mark to send archived games to the frontend. It seems like your `+archive-json` and `+move-json` arms in `/lib` are doing something very similar already.

Reply:
right! I considered tinkering with the %chess-game mark, but was nervous about what it might mess up.

and now that I put some time into converting `archive` on urbit to a `mop` `((mop @dau chess-game) lth)` I had to add extra logic to the  `%chess-archive` mark.

this might be oversharing, but when you convert a `mop` into a `list`, I'm using `tap:on`, it produces a `(list item)` where each item is `[@dau chess-game]`. because of that, I need to separate the tail from the head to plug in each `chess-game` into the new `archive-json` arm. 
_I should just make a PR and highlight that for discussion._

**do we think we'll convert the `games=(map @dau active-game-state)` into a `mop` too?**

Reply:
*Personally* I'm not too bothered by data being out of order in a `map`, if there's no very compelling reason to make it an ordered `mop`. Right now if a player had *many* games active or in their archive, I'd rather be doing all the sorting of that on the frontend where everything runs much faster than it does on Urbit. Especially if the sorting of games is just being done for presentation.

Reply:
Yeah, searching will be more efficient in the front end, but that would require downloading the whole archive to search it. Without at least a little bit of ordering on mars, that could be too much over time. 

One possibility would be to scry for all of the game-ids like with `x/games/`, which would help with size. The front end could scry for this long list of game-ids, reduce it down to a list it's interested in displaying (like the latest 16), then scry again for the data of each specific game.

But the **only** data you could sort through would be dates, and I think players will naturally want to search for @p and even event names.

Reply:
This might be "over-engineering", and just theoritical. But I'm considering using [dip:on](https://developers.urbit.org/reference/hoon/zuse/2m#dipon), which steps through a `mop` inorder, holds state, and can modify values. 

My thought is that the `dip:on` can start at the most recent games, analyze if they fit the search criteria, and keep count of the values that match. Once it hits 8 or 16 matches, it can stop and send those results. 
_dip:on probably can keep it's place, so the front end could ask for more and the the search would begin where it left off._

Once `dip:on` reaches the end of the archive it sends whatever it found.

Reply:
All the above sounds good. I'm trying a similar tactic in Telos of only running operations on a subset of all the available data.

Reply:
Should be resolved by [this comment](https://github.com/thecommons-urbit/chess/issues/65#issuecomment-1424056224) on #65; please reopen conversation if that's not the case.